packages/ng/popover2/popover.directive.ts

```diff
@@ -0,0 +1,203 @@
+			{ originX: 'start', originY: 'center' },
+			{
+				overlayX: 'end',
+				overlayY: 'center',
+			},
+		),
+		after: new ConnectionPositionPair(
+			{ originX: 'end', originY: 'center' },
+			{
+				overlayX: 'start',
+				overlayY: 'center',
+			},
+		),
+	};
+
+	opened = signal(false);
+
+	@HostBinding('attr.aria-describedby')
+	ariaDescribedBy = `popover-content-${nextId++}`;
+
+	constructor() {
+		combineLatest([toObservable(this.luPopoverOpenDelay), toObservable(this.luPopoverCloseDelay), toObservable(this.luPopoverTrigger)])
+			.pipe(
+				filter(([, , trigger]) => {
+					return trigger.includes('hover');
+				}),
+				switchMap(([openDelay, closeDelay]) => {
+					return merge(this.open$.pipe(map(() => 'open')), this.close$.pipe(map(() => 'close'))).pipe(
+						debounce((event) => {
+							return timer(event === 'open' ? openDelay : closeDelay);
+						}),
+					);
+				}),
+				takeUntilDestroyed(this.#destroyRef),
+			)
+			.subscribe((event) => {
+				if (event === 'open') {
+					this.openPopover(true);
+				} else {
+					this.#componentRef?.close();
+				}
+			});
+	}
+
+	@HostListener('mouseenter')
+	onMouseEnter() {
+		this.open$.next();
+	}
+
+	@HostListener('mouseleave')
+	onMouseLeave() {
+		this.close$.next();
+	}
+
+	@HostListener('click')
+	click(): void {
+		if (this.opened()) {
+			this.#componentRef?.close();
+		} else {
+			this.openPopover();
+		}
+	}
+
+	openPopover(disableFocusHandler = false): void {
+		if (!this.opened() && !this.luPopoverDisabled) {
+			this.opened.set(true);
+			this.#overlayRef = this.#overlay.create({
+				positionStrategy: this.#overlay
+					.position()
+					.flexibleConnectedTo(this.#elementRef)
+					.withPositions(this.customPositions || this.#buildPositions()),
+				scrollStrategy: this.#overlay.scrollStrategies.block(),
+				hasBackdrop: this.luPopoverTrigger() === 'click',
+				backdropClass: '',
+				disposeOnNavigation: true,
+			});
+			// Close on backdrop click even if backdrop is invisible
+			this.#overlayRef.backdropClick().subscribe(() => this.#componentRef.close());
+			const config: PopoverConfig = {
+				content: this.content,
+				ref: this.#overlayRef,
+				contentId: this.ariaDescribedBy,
+				triggerElement: this.#elementRef.nativeElement,
+				disableFocusManipulation: disableFocusHandler,
+			};
+			this.#componentRef = this.#overlayRef.attach(
+				new ComponentPortal(
+					PopoverContentComponent,
+					this.#vcr,
+					Injector.create({
+						providers: [{ provide: POPOVER_CONFIG, useValue: config }],
+					}),
+				),
+			).instance;
+			// On tooltip leave => trigger close
+			this.#componentRef.mouseLeave$.pipe(takeUntilDestroyed(this.#componentRef.destroyRef)).subscribe(() => this.close$.next());
+			// On tooltip enter => trigger open to keep it opened
+			this.#componentRef.mouseEnter$.pipe(takeUntilDestroyed(this.#componentRef.destroyRef)).subscribe(() => this.open$.next());
+			this.#componentRef.closed$.subscribe(() => this.opened.set(false));
```

Comment:
We should clear subscription when popover directive is destroyed as well with a `takeUntilDestroyed(this.#destroyRef)`.
I think you should `complete` all emitters in the content component `ngOnDestroy` (`mouseLeave$`, `mouseEnter$`, `closed$`).

Reply:
I didn't clear the subscription because closed$ completes on popover close but better safe than sorry indeed !

Reply:
My point was also that mouseLeave$ subscription is cleared on content destroy (good) but not on directive destroy (missing).

Reply:
Should we go with a double `takeUntilDestroyed`?